src/node_buffer.cc

```diff
@@ -835,7 +835,9 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 
   size_t cmp_length = MIN(obj_a_length, obj_b_length);
 
-  int32_t val = memcmp(obj_a_data, obj_b_data, cmp_length);
+  int32_t val = 0;
```

Comment:
Btw, while you are at it: what is the reason for `val` to be `int32_t`? `memcmp()` returns an `int` afaik. While it is fine for all the available compilers that I am aware of, what would happen in a case if `int` is a 64-bit?